src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,69 @@
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvModel;
+
+import com.troubleshooting.model.EnvsModel;
+import com.troubleshooting.model.LogsModel;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+// Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+@Component
+public class ContainerDataRetrieval implements IDataRetrieval {
```

Comment:
As a rule of thumb if you don't plan for this class to be extended, declare it final

Reply:
is this a rule of thumb for all classes?

Reply:
Take into consideration that if it's not final it can be extended.

Reply:
why is extended a class an issue?

Reply:
if write a code that depends on that class, someone may provide a behaviour that do not expect. If you plan to let. others extend the class that is OK.